### PR TITLE
X.H.ManageDocks: React to strut updates of override_redirect docks

### PR DESCRIPTION
**Pull request by liskin**
### Description

We only requested PropertyChange events from docks in the `manageDocks` manageHook, but that only gets called for normal windows, not override_redirect ones. Therefore, xmobar in its default configuration wouldn't get its struts refreshed on changes. This resulted in gaps not updating after xmobar repositions itself on xrandr changes.

If one wanted to handle that repositioning in xmonad, it was possible to configure xmobar with `overrideRedirect = False`, which is meant for window managers with [proper EWMH stacking order support][1], so in xmonad it resulted in xmobar covering fullscreen windows. That can be worked around by adding `checkDock --> doLower` to manageHook, but it starts to smell of too many workarounds.

[1]: https://specifications.freedesktop.org/wm-spec/wm-spec-1.5.html#STACKINGORDER

The fix is to request PropertyChange events for all windows that we treat as docks.

Related: https://github.com/xmonad/xmonad-contrib/pull/406
Related: https://github.com/xmonad/xmonad-contrib/pull/490

### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [X] I updated the `CHANGES.md` file

  - n/a I updated the `XMonad.Doc.Extending` file (if appropriate)

### CODE REVIEW

File: XMonad/Util/ExtensibleState.hs [hunk 1]
```diff
@@ -121,8 +123,11 @@ remove :: (ExtensionClass a, XLike m) => a -> m ()
 remove wit = modifyStateExts $ M.delete (show . typeOf $ wit)
 
 modified :: (ExtensionClass a, Eq a, XLike m) => (a -> a) -> m Bool
-modified f = do
+modified = modifiedM . (pure .)
```
Comment (geekosaur) [inline XMonad/Util/ExtensibleState.hs:126]: Arguably this refactor should be a separate PR.

**Review by geekosaur — APPROVED** [review]
Other than that (and a similar feeling about the `deleteFromStru[c]tCache` correction), this seems okay.

File: XMonad/Util/ExtensibleState.hs (hunk 1 above)
Comment (liskin) [reply XMonad/Util/ExtensibleState.hs:126]: I can definitely turn this into a commit on its own but a separate PR seems excessive.

Comment (liskin) [reply XMonad/Util/ExtensibleState.hs:126]: > Other than that (and a similar feeling about the `deleteFromStru[c]tCache` correction), this seems okay.

I split both into separate commits.
